cockatrice/src/tab_supervisor.cpp

```diff
@@ -559,3 +561,19 @@ bool TabSupervisor::getAdminLocked() const
         return true;
     return tabAdmin->getLocked();
 }
+
+void TabSupervisor::processNotifyUserEvent(const Event_NotifyUser &event)
+{
+    QString notificationTitle("Notification");
+    QString notificationMessage;
+    switch ((Event_NotifyUser::NotificationType) event.type()) {
+        case Event_NotifyUser::PROMOTIONSUCCESS: QMessageBox::information(this, tr("Success"), tr("Successfully promoted user.")); break;
```

Comment:
This isn't the right use for this. This data should be contained in the `Response` to the promotion action.

Look at the pr for adding registration protocol if you want an example of how to add new responses with commands.